jupyter_console/interactiveshell.py

```diff
@@ -597,3 +637,21 @@ def init_history(self):
         self.history_manager = ZMQHistoryManager(client=self.client)
         self.configurables.append(self.history_manager)
 
+    def get_source_and_reset(self):
```

Comment:
Since this is two simple lines of code and it's called in two places, I think it would be clearer to inline it.